apps/teams/tasks.py

```diff
@@ -18,9 +20,11 @@ def send_invitation_accepted_notification(invitation_id):
 @shared_task
 def delete_team_async(team_id):
     team = Team.objects.get(id=team_id)
-
-    # get required info for emails prior to deleting team info
-    team_name = team.name
     admin_emails = get_admin_emails_with_delete_permission(team)
+    team_name = team.name
+    chunk_size = 50
+    chunked_emails = chunk_list(admin_emails, chunk_size)
     delete_object_with_auditing_of_related_objects(team)
-    send_team_deleted_notification(team_name, admin_emails)
+    for chunk_emails in chunked_emails:
+        with current_team(team):
```

Comment:
I think this should be guarding the `delete_object_with_auditing_of_related_objects` call. I don't think it's needed for the email sending.